pkg/cmd/infra.go

```diff
@@ -0,0 +1,373 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"regexp"
+	"io/ioutil"
+
+	"github.com/spf13/cobra"
+)
+
+// NewInfraCmd returns a new infra command
+func NewInfraCmd(targetReader TargetReader) *cobra.Command {
+	return &cobra.Command{
+		Use:          "infra [(orphan)] [(list)]",
+		Short:        "Manage shoot infra resources\n",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			target := targetReader.ReadTarget(pathTarget)
+			if !CheckShootIsTargeted(target) {
+				return errors.New("no shoot targeted")
+			}
+			if len(args) < 2 || len(args) > 2 {
+				return errors.New("command must be in the format: infra [orphan] [list]")
+			}
+			switch args[0] {
+			case "orphan":
+				switch args[1] {
+				case "list":
+					var rs []string
+					pathTerraformState := downloadTerraformFiles("infra", targetReader) + "/terraform.tfstate"
+					buf, err := ioutil.ReadFile(pathTerraformState)
+					if err != nil || len(buf) < 64 {
+						fmt.Println("Could not read terraform.tfstate: " + pathTerraformState)
+						os.Exit(2)
+					}
+					terraformstate := string(buf)
+
+					shoot, err := FetchShootFromTarget(target)
+					checkError(err)
+					infraType := shoot.Spec.Provider.Type
+
+					switch infraType {
+						case "aws":
+							rs = getAWSInfraResources()
+						case "gcp":
+						case "azure":
+							rs = getAzureInfraResources()
+						case "alicloud":
+						case "openstack":
```

Comment:
what happens in case of gcp, alicloud and openstack? Currently not implemented? Is the user aware?